### PR TITLE
Update for JLab 1.0.0

### PR DESCRIPTION
**Pull request by jasongrout**
Don't merge just yet, just keep this as a branch in this repo.

### CODE REVIEW

File: {{cookiecutter.extension_name}}/package.json [hunk 1]
```diff
@@ -19,22 +19,23 @@
   ],
   "main": "lib/index.js",
   "types": "lib/index.d.ts",
+  "style": "style/index.css",
```
Comment (vidartf) [inline {{cookiecutter.extension_name}}/package.json]: Also sideEffects?

File: {{cookiecutter.extension_name}}/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "watch": "tsc -w"
   },
   "dependencies": {
-    "@jupyterlab/application": "^1.0.0-rc.0"
+    "@jupyterlab/application": "^1.0.0"
```
Comment (SimonBiggs) [inline {{cookiecutter.extension_name}}/package.json]: Wow... beautiful moment :)

**Comment by SgtPooki** [comment]
@jasongrout @ellisonbg @blink1073 I ran through the tutorial a few weeks ago. I noticed these changes and decided to update my plugin since it's still fairly simple. 

I ran into a change that doesn't seem to be documented anywhere: `app.shell.addToMainArea` has been changed to `app.shell.add`. The only mention of addToMainArea I can find is at https://github.com/jupyterlab/jupyterlab/commit/ed8692d6b3367628fe4505bfd4b9818af2d5e7b0

Maybe my github foo needs work, but I wanted to let you know either way.

**Comment by jasongrout** [comment]
Good point. Would you like to add it to the list at https://jupyterlab.readthedocs.io/en/stable/getting_started/changelog.html#extension-development-changes ? Somehow we missed that. The relevant source is at https://github.com/jupyterlab/jupyterlab/blob/master/docs/source/getting_started/changelog.rst

**Comment by jasongrout** [comment]
Relevant commit is https://github.com/jupyterlab/jupyterlab/commit/c59e46b814c5f34e317cd482fed09d53c18d4365, in PR https://github.com/jupyterlab/jupyterlab/pull/5845

**Comment by jasongrout** [comment]
We do have the note "We have renamed the plugin type from JupyterLabPlugin to JupyterFrontEndPlugin. The application arg is also renamed from JupyterLab to JupyterFrontEnd and some its functionality has been moved to a separate ILabShell plugin (#5845, #5919)", but it doesn't mention anything about addToMainArea changing. Likely that should be a separate bullet point to emphasize it, though.

**Comment by jasongrout** [comment]
> Would you like to add it to the list at

Done in https://github.com/jupyterlab/jupyterlab/pull/6899

**Pull request merged**